.github/workflows/build_and_test.yml

```
@@ -0,0 +1,56 @@
+# Workflow for building and testing eventuals on macOS, Ubuntu and Windows.  
+name: Build and Run all tests
+
+on:
+  push:
+    branches:
+      - "**"
+    paths-ignore:
+      - "**.md"
+  
+jobs:
+  build-and-test:
+    name: Build and Test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+
+    steps:
+      # Checkout the repository under $GITHUB_WORKSPACE.
+      - uses: actions/checkout@v2
+
+      - name: Check for bazel
+        run: bazel --version
+
+      - name: Build step on macOS or Ubuntu
+        if: ${{ matrix.os == 'macos-latest' || matrix.os == 'ubuntu-latest' }}
+        run: bazel build --spawn_strategy=local --copt="-g" --strip="never" :eventuals
+
+      - name: Debug Build using tmate (if failure)
+        if: ${{ failure() }}
+        uses: mxschmitt/action-tmate@v3    
+
+      - name: Build step on Windows
+        if: ${{ matrix.os == 'windows-latest' }}
+        run: bazel build --spawn_strategy=local --strip="never" :eventuals  
+
+      - name: Debug Build using tmate (if failure)
+        if: ${{ failure() }}
+        uses: mxschmitt/action-tmate@v3
+
+      - name: Test step on macOS or Ubuntu
+        if: ${{ matrix.os == 'macos-latest' || matrix.os == 'ubuntu-latest' }} 
+        run: bazel test --spawn_strategy=local --copt="-g" --strip="never" --test_output=errors test:eventuals --test_arg=--gtest_shuffle --test_arg=--gtest_repeat=100
+
+      - name: Debug Test using tmate (if failure)
+        if: ${{ failure() }}
+        uses: mxschmitt/action-tmate@v3
+
+      - name: Test step on Windows
+        if: ${{ matrix.os == 'windows-latest' }} 
+        run: bazel test --spawn_strategy=local --strip="never" --test_output=errors test:eventuals --test_arg=--gtest_shuffle --test_arg=--gtest_repeat=100
```

Does it work if we do `-c dbg` for Windows? Then we can have a single Build step and single Test step with no `if`. Also let's wrap these long lines, I've been meaning to do that for a while. See suggestion below for both!
 
```suggestion
      - name: Build
        run: bazel build \
               --spawn_strategy=local \
               -c dbg \
               --strip="never" \
               :eventuals

      - name: Test
        run: bazel test \
               --spawn_strategy=local \
               -c dbg \
               --strip="never" \
               --test_output=errors \
               test:eventuals \
               --test_arg=--gtest_shuffle \
               --test_arg=--gtest_repeat=100
```